src/Observable.ts

```diff
@@ -105,8 +105,26 @@ export class Observable<T> extends BaseObservable<T> {
         })
     }
 
-    scan<U>(scanner: (values: T[]) => U, historyLength = 2, defaultValue = null): Observable<U> {
-        throw new Error('not implemented')
+    scan(historyLength = 2, defaultValue = null): Observable<T[]> {
+        const history: T[] = defaultValue !== null
+            ? Array(historyLength).fill(defaultValue)
+            : []
+        return new Observable<T[]>(observer => {
+            const subscription = this.subscribe(
+                value => {
+                    if (observer.closed) {
```

Comment:
Ok, jak rozumiem to we wszystkich napisanych przeze mnie funkcjach usunąć ten warunek?